docs/source/migration.md

```diff
@@ -0,0 +1,140 @@
+### Version  migrations
+
+
+## `v0.3.0` -> `v1.0.0` migration
+
+`v0.3.0` used a very common name as a CRD kind. In `v1.0.0` this issue was solved by using less common kind
+which is easier to disambiguate.
+This change is backward incompatible, which means manual migration is needed.
+
+This procedure involves having two CRDs registered at the same time. For a short period, we will detach Scylla pods
+from Scylla Operator, to ensure that nothing is garbage collected when Scylla Operator is upgraded.
+Compared to upgrade guide where full deletion is requested, this procedure shouldn't cause downtimes.
```

Comment:
which upgrade guide? link?
if you are referring to and old version, no need to mention it